src/ncbi_gene/cli.py

```diff
@@ -12,15 +12,29 @@
 
 
 @app.callback()
-def callback(version: bool = typer.Option(False, "--version", is_eager=True),
-):
+def callback(
+        version: bool = typer.Option(False, "--version", is_eager=True),
+        verbose: int = typer.Option(True, help="Verbosity level: 0 (warning), 1 (info), 2+ (debug)"),
+        quiet: bool = typer.Option(False, help="Suppress logging output")
+    ):
     """NCBI Gene CLI."""
     if version:
         from ncbi_gene import __version__
         typer.echo(f"NCBI Gene version: {__version__}")
         raise typer.Exit()
 
 
+    if verbose:
+        logger.setLevel(level=logging.DEBUG)
+    elif verbose is None:
+        logger.setLevel(level=logging.INFO)
+    else:
+        logger.setLevel(level=logging.WARNING)
+    if quiet:
+        logger.setLevel(level=logging.ERROR)
+    logger.info(f"Logger {logger.name} set to level {logger.level}")
+
+
```

Comment:
ohhhh, nice, you should put this change into the cookiecutter-monarch-ingest repo